src/components/BreadCrumb/BreadCrumb.test.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import BreadCrumb from './BreadCrumb';
+import { BreadCrumbProps } from './BreadCrumb';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+describe('BreadCrumb Component', () => {
+  const renderComponent = (props: BreadCrumbProps) =>
+    render(<BreadCrumb {...props} />);
+
+  test('renders all breadcrumb items', () => {
+    const items = [
+      { label: 'Home', href: '/' },
+      { label: 'Services', href: '/services' },
+      { label: 'Plumbing', href: '/services/plumbing' },
+    ];
+
+    renderComponent({ items });
+
+    items.forEach((item) => {
+      expect(screen.getByText(item.label)).toBeInTheDocument();
+    });
+  });
+
+  test('highlights the last breadcrumb item as bold', () => {
+    const items = [
+      { label: 'Home', href: '/' },
+      { label: 'Services', href: '/services' },
+      { label: 'Plumbing' },
+    ];
+
+    renderComponent({ items });
+
+    const lastItem = screen.getByText('Plumbing').closest('li');
+
+    expect(lastItem).toBeInTheDocument();
+
+    expect(lastItem).toHaveStyle('font-weight: bold');
+  });
+
+  test('renders correctly with an empty items array', () => {
+    const { container } = render(<BreadCrumb items={[]} />);
+
+    expect(container.firstChild).toBeNull();
+  });
+
+  test('handles edge case with only one item', () => {
+    const items = [{ label: 'Home', href: '/' }];
+
+    renderComponent({ items });
+
+    expect(screen.getByText('Home')).toBeInTheDocument();
+  });
+
+  test('navigates to the correct href when a breadcrumb item is clicked', () => {
+    const items = [
+      { label: 'Home', href: '/' },
+      { label: 'Services', href: '/services' },
+      { label: 'Plumbing', href: '/services/plumbing' },
+    ];
+    render(<BreadCrumb items={items} />);
+
+    const originalLocation = window.location;
+    Object.defineProperty(window, 'location', {
+      value: { href: '' },
+      writable: true,
+    });
+
+    fireEvent.click(screen.getByText('Services'));
+
+    expect(window.location.href).toBe('/services');
+
+    window.location = originalLocation;
```

Comment:
every test case should have the expect statement as the last statement of the test case. after validation, nothing should be added.